src/assisted_test_infra/test_infra/controllers/minikube.py

```diff
@@ -0,0 +1,242 @@
+import logging
+import os
+import time
+
+import waiting
+
+
+class MinikubeController:
+    """Manage minikube commands / config changes via k8_client
+    Allow to modify configmap and verify changes
+    Allow to stop/start resources running on minukube plus waiters
+    for actions completed till pods deleted or created.
+    Enable metrics to check memory leaks and cpu usage - TBD
+    In order to access to data resource we:
+    _get_resources_by_kind -> this is the kind of resource
+    _get_obj_resources_by_name -> get the obj resource name
+    """
+
+    CONFIGMAP = "ConfigMap"
+    DEPLOYMENT = "Deployment"
+    REPLICASET = "ReplicaSet"
+    POD = "Pod"
+    SERVICE = "Service"
+
+    def __init__(self, k8_client):
```

Comment:
As @adriengentil mentioned, Please add type and return type annotations to all of the declared functions